src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs

```diff
@@ -110,46 +110,60 @@ public override bool RunTask ()
 		protected override string GenerateCommandLineCommands ()
 		{
 			var cmd = GetCommandLineBuilder ();
-			cmd.AppendFileNameIfNotNull (ApiXmlInput);
 
-			if (OnlyRunXmlAdjuster)
-				cmd.AppendSwitch ("--only-xml-adjuster");
-			cmd.AppendSwitchIfNotNull ("--xml-adjuster-output=", XmlAdjusterOutput);
+			string responseFile = Path.Combine (OutputDirectory, "generator.rsp");
+			using (var sw = new StreamWriter (responseFile, append: false, encoding: MonoAndroidHelper.UTF8withoutBOM)) {
+
+				if (OnlyRunXmlAdjuster)
+					sw.WriteLine ("--only-xml-adjuster");
+				if (!string.IsNullOrEmpty (XmlAdjusterOutput))
+					sw.WriteLine ($"--xml-adjuster-output=\"{XmlAdjusterOutput}\"");
+
+				if (!string.IsNullOrEmpty (CodegenTarget))
+					sw.WriteLine ($"--codegen-target={CodegenTarget}");
+				if (!string.IsNullOrEmpty (OutputDirectory))
+					sw.WriteLine ($"--csdir=\"{OutputDirectory}\"");
+				if (!string.IsNullOrEmpty (EnumDirectory))
+					sw.WriteLine ($"--enumdir=\"{EnumDirectory}\"");
+				if (!string.IsNullOrEmpty (EnumMetadataDirectory))
+					sw.WriteLine ($"--enummetadata=\"{EnumMetadataDirectory}\"");
+				if (!string.IsNullOrEmpty (AssemblyName))
+					sw.WriteLine ($"--assembly={AssemblyName}");
+
+				if (!NoStdlib) {
+					string fxpath = MonoAndroidFrameworkDirectories.Split (';').First (p => new DirectoryInfo (p).GetFiles ("mscorlib.dll").Any ());
+					sw.WriteLine ($"--ref=\"{Path.Combine (Path.GetFullPath (fxpath), "mscorlib.dll")}\"");
+				}
 
-			cmd.AppendSwitchIfNotNull ("--codegen-target=", CodegenTarget);
-			cmd.AppendSwitchIfNotNull ("--csdir=", OutputDirectory);
-			cmd.AppendSwitchIfNotNull ("--enumdir=", EnumDirectory);
-			cmd.AppendSwitchIfNotNull ("--enummetadata=", EnumMetadataDirectory);
-			cmd.AppendSwitchIfNotNull ("--assembly=", AssemblyName);
+				if (ReferencedManagedLibraries != null)
+					foreach (var lib in ReferencedManagedLibraries)
+						sw.WriteLine ($"--ref=\"{Path.GetFullPath (lib.ItemSpec)}\"");
+				if (AnnotationsZipFiles != null)
+					foreach (var zip in AnnotationsZipFiles)
+						sw.WriteLine ($"--annotations=\"{Path.GetFullPath (zip.ItemSpec)}\"");
 
-			if (!NoStdlib) {
-				string fxpath = MonoAndroidFrameworkDirectories.Split (';').First (p => new DirectoryInfo (p).GetFiles ("mscorlib.dll").Any ());
-				cmd.AppendSwitchIfNotNull ("--ref=", Path.Combine (Path.GetFullPath (fxpath), "mscorlib.dll"));
-			}
-			
-			if (ReferencedManagedLibraries != null)
-				foreach (var lib in ReferencedManagedLibraries)
-					cmd.AppendSwitchIfNotNull ("--ref=", Path.GetFullPath (lib.ItemSpec));
-			if (AnnotationsZipFiles != null)
-				foreach (var zip in AnnotationsZipFiles)
-					cmd.AppendSwitchIfNotNull ("--annotations=", Path.GetFullPath (zip.ItemSpec));
+				foreach (var tf in transform_files)
+					sw.WriteLine ($"--{tf.Item2}={tf.Item1}");
```

Comment:
If `transform_files` contains filenames, then we should probably quote this *entire* string:

```csharp
sw.WriteLine ($"\"--{tf.Item2}={tf.Item1}\"");
```